api-server/src/server/boot/donate.js

```diff
@@ -174,7 +174,7 @@ export default function donateBoot(app, done) {
       .then(() => res.status(200).json({ isDonating: true }))
       .catch(err => {
         log(err.message);
-        return res.status(500).json({
+        return res.status(400).json({
```

Comment:
@ojeytonwilliams Why should this be `400`? This will fail if a DB write fails - almost entirely server -related.

Reply:
I agree, this should be a 500 error. I've updated the issue to reflect that.

It should be a 400 when the client attempts to set `isDonating` if the user is already. I don't know if that's even possible with the loopback API.

Reply:
It is already available in the user state, when the page loads, etc.